services/httpd/handler.go

```diff
@@ -517,6 +518,10 @@ func (h *Handler) serveWrite(w http.ResponseWriter, r *http.Request, user *meta.
 	}
 
 	h.statMap.Add(statPointsWrittenOK, int64(len(points)))
+
+	if w.Header().Get("Content-Encoding") == "gzip" {
```

Comment:
I think it would be better to remove the header management from here completely and do it in the `gzipResponseWriter`. Something like this:

```
func (w *gzipResponseWriter) WriteHeader(code int) {
    if code != http.StatusNoContent {
        w.Header().Set("Content-Encoding", "gzip")
    }
    w.ResponseWriter.WriteHeader(code)
}
```

Then you can remove all of the locations where `Content-Encoding` is set or removed.